File: openapi-templates/go/api_extra/application_create.go
// Get or create a new application.
func (application *Application) GetOrCreate(
	ctx context.Context,
	applicationIn models.ApplicationIn,
	o *ApplicationCreateOptions,
) (*models.ApplicationOut, error) {
	queryMap := map[string]string{
		"get_if_exists": "true",
	}
	headerMap := map[string]string{}

	var err error
	if o != nil {
		serializeParamToMap("idempotency-key", o.IdempotencyKey, headerMap, &err)
		if err != nil {
			return nil, err
		}
	}

	return executeRequest[models.ApplicationIn, models.ApplicationOut](
		ctx,
		application.client,
		"POST",
		"/api/v1/app",
		nil,
		queryMap,
		headerMap,
		&applicationIn,
	)
}

File: codegen-templates/javascript/api_extra/application_create.ts
/** Get the application with the UID from `applicationIn`, or create it if it doesn't exist yet. */
public getOrCreate(
  applicationIn: ApplicationIn,
  options?: ApplicationCreateOptions
): Promise<ApplicationOut> {
  const request = new SvixRequest(HttpMethod.POST, "/api/v1/app");

  request.setQueryParam("get_if_exists", true);
  request.setHeaderParam("idempotency-key", options?.idempotencyKey);
  request.setBody(ApplicationInSerializer._toJsonObject(applicationIn));

  return request.send(this.requestCtx, ApplicationOutSerializer._fromJsonObject);
}
